hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -373,6 +375,8 @@ public RegisteredCommand register(
 
     String dnsName;
     String networkLocation;
+    String ipAddress = datanodeDetails.getIpAddress();
+    String hostName = datanodeDetails.getHostName();
```

Comment:
Please also be very careful. In my impression, in some of my test, datanodeDetails.getHostName() might give you IP instead of host name. Please test if your approach does not have bugs for this case as well.

Reply:
@sokui can you elaborate when you see an IP address being returned for hostname?

Reply:
@kerneltime it has been so long time, I could not remember. But my testing environment is Kubernetes. I think when we deployed ozone, I see this happens a couple of times. But not sure what is the exact condition.